libraries/dagster-delta/dagster_delta/_db_io_manager/utils.py

```diff
@@ -132,16 +138,16 @@ def hourly_delta(self) -> int:
 
     def is_consecutive(self) -> bool:
         """Checks whether the provided start dates of each partition timewindow is consecutive"""
-        return (
-            len(
-                {
-                    pdi(self.start).add(hours=self.hourly_delta * i)
-                    for i in range(date_diff(self.start, self.end).in_days() + 1)
-                }
-                - {pdi(d.start) for d in self._partitions},
-            )
-            == 1
-        )
+        total_hours = date_diff(self.start, self.end).in_hours()
+        num_steps = int(total_hours / self.hourly_delta)
+
+        expected_starts = {
+            pdi(self.start).add(hours=self.hourly_delta * i) for i in range(num_steps)
+        }
```

Comment:
```suggestion
        expected_starts = {
            pdi(self.start).add(hours=self.hourly_delta * i) for i in range(len(self._partitions))
        }
```
Wouldn't this be simpler? :)

Reply:
it would be, but this would make the `test_multi_partitioned_to_non_partitioned_asset` and `test_generate_partition_dimensions_color_varying` test fail ;)